### PR TITLE
WAR-2242: Reduce the Warrior Setup and Startup Time.

### PR DESCRIPTION
**Pull request by vkotakon-warrior**

### CODE REVIEW

File: warrior/Tools/tc_generator/tc_generator [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+import site
+import sys
+from xml.etree import ElementTree as et
+
+# Argparser to add command line argumets
+parser = argparse.ArgumentParser()
+parser.add_argument('-t', '--type', choices=['cli','snmp','netconf', 'rest'],\
+ help="type argument support test cases", required=True)
+parser.add_argument('-tc','--tc_name', help="Name of the the test case ", required=True)
+parser.add_argument('--ip', help="Ip address of the device")
```
Comment (santhosh-warrior) [inline warrior/Tools/tc_generator/tc_generator]: Not Consistent with other options.
You need add -ip and --ipaddr  both - and "--" option.

File: warrior/Tools/tc_generator/templates/netconf_test_case_template.xml [hunk 2]
```diff
@@ -0,0 +1,204 @@
+<?xml version="1.0" ?>
+<Testcase>
+	<Details>
+		<Name>tc_netconf_sanity</Name>
+		<Title>tc_netconf_sanity</Title>
+		<Engineer>venkat</Engineer>
```
Comment (santhosh-warrior) [inline warrior/Tools/tc_generator/templates/netconf_test_case_template.xml]: Add Warrior User  as Engineer
don't hard code your user name.

File: warrior/Tools/tc_generator/templates/rest_data_file_template.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" ?>
+<credentials>
+	<system name="http_bin_1">
+		<url>http://httpbin.org</url>
+		<username>venkat</username>
```
Comment (santhosh-warrior) [inline warrior/Tools/tc_generator/templates/rest_data_file_template.xml]: remove the user name

File: warrior/Tools/tc_generator/templates/rest_test_case_template.xml [hunk 4]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" ?>
+<Testcase>
+	<Details>
+		<Name>tc_rest_verify</Name>
+		<Title>To test verify api response keyword </Title>
+		<Engineer>venkatadhri kotakonda</Engineer>
```
Comment (santhosh-warrior) [inline warrior/Tools/tc_generator/templates/rest_test_case_template.xml]: remove the user name

File: warrior/Tools/tc_generator/templates/cli_test_case_without_dip_template.xml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                        <context>positive</context>
+                        <impact>impact</impact>
+                </step>
+            <step Driver="cli_driver" Keyword="send_commands_by_testdata_title" TS="4">
+                        <Arguments>
+                                <argument name="system_name" value="NE1" />
+                                <argument name="title" value="execute_id_command" />
+                        </Arguments>
+                        <onError action="abort" />
+                        <Description>connect</Description>
```
Comment (santhosh-warrior) [inline warrior/Tools/tc_generator/templates/cli_test_case_without_dip_template.xml]: Add proper description in the templates.

File: warrior/Tools/tc_generator/templates/netconf_test_data_template.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+<configs>
+
+  <config_data name='addUserConfig'>
+     <config>
+      <users xmlns="urn:fujitsu:params:xml:ns:yang:usersecurity">
+      <user>
+        <username>warrioruser</username>
+        <group>level-4</group>
+        <adminState>allow</adminState>
+      <password>JUnk123</password>
```
Comment (apathapa-warrior) [inline warrior/Tools/tc_generator/templates/netconf_test_data_template.xml]: Remove the static value .

File: warrior/Tools/tc_generator/templates/netconf_test_data_template.xml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+      </user>
+    <user>
+        <username>sample</username>
+        <group>level-4</group>
+        <adminState>allow</adminState>
+      <password>JUnk123</password>
```
Comment (apathapa-warrior) [inline warrior/Tools/tc_generator/templates/netconf_test_data_template.xml]: Same as above

**Pull request closed without merging**